.github/workflows/build-and-test.yaml

```diff
@@ -51,3 +51,17 @@ jobs:
           files: ${{ steps.test.outputs.coverage-report-files }}
           fail_ci_if_error: false
           verbose: true
+          flags: total
+
+      - name: Move directory
+        run: |
+          mv lcov lcov-total || mkdir lcov-total
+          mv coveragepy coveragepy-total || mkdir coveragepy-total
+
+      - name: Cache coverage artifacts
+        uses: actions/cache@v2
+        with:
+          path: |
+            ./lcov-total
+            ./coveragepy-total
+          key: coverage-${{ github.sha }}
```

Comment:
Same as https://github.com/autowarefoundation/autoware.universe/pull/384/files#r808805929.
Why do you not run this in the `colcon-test` workflow?

Reply:
Do you have a suggestion to include this in the `colcon-test`?
It is possible, assuming that the coverage is cached in all pull request bases.
In other words, we need an `on: push` to `build-and-test` workflow.

Reply:
I'm sorry, I couldn't understand it.
There is already the `on: push` condition.